pip/wheel.py

```diff
@@ -37,6 +37,12 @@
 from pip.utils.temp_dir import TempDirectory
 from pip.utils.ui import open_spinner
 
+try:
+    from concurrent.futures.process import ProcessPoolExecutor
```

Comment:
> The vendored version could fail on unreleased newer python versions.

Which would warrant a new release of pip, which is fine. You can't support something that doesn't exist when you built it. No laptop supported USB-C before it was standardised, even though OEMs knew how it would _almost_ be.

OTOH, If the user install a release of concurrent.futures that pip has not been tested with and pip would not work - it is pip's fault and if you think about it, it'll be impossible to use pip if it has such a try-except to recover from a situation.

> Trying to maintain python core libraries is just duplicating efforts.

pip simply vendors versions of these libraries that it knows it works with. They are all still separate projects in their own right and they just happen to be used by pip and because pip is special (for various reasons that I won't go into), it vendors its dependencies.

---

Point being - any third party library that pip imports _must_ be a vendored within pip and only that vendored library should be used.